homeassistant/components/neptune_apex/__init__.py

```diff
@@ -0,0 +1,80 @@
+"""The Neptune Apex integration."""
+import asyncio
+from datetime import timedelta
+import logging
+
+import async_timeout
+from pynepsys import Apex
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+
+PLATFORMS = ["sensor", "light"]
+
+NEPTUNE_APEX = "neptune_apex"
+NEPTUNE_APEX_COORDINATOR = "neptune_apex_coordinator"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Neptune Apex component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Neptune Apex from a config entry."""
+    apex = Apex(entry.data["host"], entry.data["username"], entry.data["password"])
+    hass.data[NEPTUNE_APEX] = apex
+
+    async def async_update_data():
+        """Fetch data for all outlets and probes at once."""
+        try:
+            async with async_timeout.timeout(10):
+                await apex.fetch_current_state()
+        except Exception as err:
```

Comment:
Is there a more specific exception we can catch? What exception do we expect?

Reply:
The library does not specify exceptions, but this causes a poll so based on the code we'd expect network issues, possible parse issues on bad data. Should we pass them up? I had assumed the component should absorb them and just generically fail while the Apex was unhappy for any reason.

Reply:
We try to avoid catching broad exceptions in non critical places. It would be best if the library caught the expected exceptions and raise library specific exceptions when needed.

Reply:
That makes sense. I will add that to the next version of pynepsys.

Reply:
https://github.com/jpelzer/pynepsys/issues/1

Reply:
Now catches ApexException